src/plan_expand_hypertable.c

```diff
@@ -51,56 +109,71 @@ collect_quals_walker(Node *node, CollectQualCtx *ctx)
 		{
 			Node	   *qual = (Node *) lfirst(lc);
 			RestrictInfo *restrictinfo;
+			Relids		relids;
+			bool		belongs_to_cur_rel = false;
+
+
+			if (IsA(qual, BoolExpr))
+				if (contains_chunk_exclusion_func(qual))
+					ereport(ERROR,
+							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							 errmsg("chunks_in function needs to be on the top level")));
 
-			Relids		relids = pull_varnos(qual);
+			relids = pull_varnos(qual);
 
-			if (bms_num_members(relids) != 1 || !bms_is_member(ctx->rel->relid, relids))
+			belongs_to_cur_rel = bms_num_members(relids) == 1 && bms_is_member(ctx->rel->relid, relids);
+
+			if (is_chunk_exclusion_func(qual))
+			{
+				FuncExpr   *func_expr = (FuncExpr *) qual;
+
+				/* validation */
+				if (ctx->chunk_exclusion_func != NULL && belongs_to_cur_rel)
+					ereport(ERROR,
+							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							 errmsg("only one chunks_in call is allowed per hypertable")));
+
+				Assert(func_expr->args->length == 2);
+				if (!IsA(linitial(func_expr->args), Var))
+					ereport(ERROR,
+							(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							 errmsg("first parameter for chunks_in function needs to be a relation name or alias")));
+
+				if (belongs_to_cur_rel)
+					ctx->chunk_exclusion_func = func_expr;
```

Comment:
I think if you remove the list entry for this from `f->quals` here, index-only scans will work again.